### DISCUSSION TITLE
Does DesktopChildSiteBridge support xaml content?

### DISCUSSION THREAD

**Discussion by sudongg** [microsoft/WindowsAppSDK#2927] [General]
I'm looking at the WindowsAppSDK-Samples/Samples/Content/cpp-win32/Squares sample and I want to know if besides Composition , Virtual , is it supported to create xaml content like Button, TextBlock etc.

**Comment by DarranRowe** [comment]
Yes and no.
Because this is using AppWindow then it cannot directly use Xaml content. However, [Xaml Islands](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.hosting.windowsxamlmanager?view=winrt-22621) exists. While this feature isn't part of the WinUI 3 release versions, it is available in the experimental versions. 
But this doesn't have the same functionality, there are some things which are just not available, and since the Xaml compiler isn't involved, you are going to have to write a few things by hand. It is possible though.
